arches/app/views/api.py

```diff
@@ -380,26 +389,34 @@ def post(self, request, resourceid=None):
             if user_can_edit_resources(user=request.user):
                 data = JSONDeserializer().deserialize(request.body)
                 reader = JsonLdReader()
-                reader.read_resource(data)
+                if slug is not None:
+                    graphid = models.GraphModel.objects.get(slug=slug).pk
+                reader.read_resource(data, graphid=graphid)
                 if reader.errors:
                     response = []
                     for value in reader.errors.itervalues():
                         response.append(value.message)
-                    return JSONResponse(data, indent=indent, status=400, reason=response)
+                    return JSONResponse({"error": response}, indent=indent, status=400)
                 else:
                     response = []
                     for resource in reader.resources:
                         with transaction.atomic():
                             resource.save(request=request)
+                        print resource.resourceinstanceid
                         response.append(JSONDeserializer().deserialize(
                             self.get(request, resource.resourceinstanceid).content))
                     return JSONResponse(response, indent=indent, status=201)
             else:
                 return JSONResponse(status=403)
         except Exception as e:
-            return JSONResponse(status=500, reason=e)
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            formatted = traceback.format_exception(exc_type, exc_value, exc_traceback)
+            if len(formatted):
```

Comment:
can we remove/comment this whole block?

Reply:
I've wrapped the code as I suggested. Is this a suitable change?

Reply:
This one OTOH is more important. The response is meaningless without what actually happened.

Reply:
mode=DEV is a good solution, :+1: to @benosteen